devices.js

```diff
@@ -15444,6 +15444,18 @@ const devices = [
             await configureReporting.onOff(endpoint);
         },
     },
+
+    // BYUN
+    {
+        zigbeeModel: ['Windows switch  '],
+        model: 'M415-6C',
+        vendor: 'BYUN',
+        description: 'Smoke sensor',
+        supports: 'Smoke alarm',
+        fromZigbee: [fz.byun_smoke_on, fz.byun_smoke_off],
```

Comment:
Can you try with `fromZigbee: [fz.ias_smoke_alarm_1]` instead here?

Reply:
I conducted a test for 48 hours, tried low supply voltage, case tamper, test and real triggers: the sensor sends only two types of clusters `ssIasZone` and `pHMeasurement`, when triggered and restored, respectively.I tried with `ias_smoke_alarm_1`, this allows you to get a smoke alarm, but then it returns a lot of arguments not related to this device (enrolled, tamper, battery_low,  supervision_reports, restore_reports, trouble, ac_status).

It seems to me that information that is deceiving is not worth showing (battery, tamper so on), so I made my own converter. 
also the proposed converter allows to process the smoke sensor restore message.

Reply:
I conducted a test, tried low supply voltage, case tamper, test and real triggers: the sensor sends only two types of clusters `ssIasZone` and `pHMeasurement`, when triggered and restored, respectively. [pastebin](https://pastebin.com/7M6Lzxif)
I tried with `ias_smoke_alarm_1`, this allows you to get a smoke alarm, but then it returns a lot of arguments not related to this device (enrolled, tamper, battery_low, supervision_reports, restore_reports, trouble, ac_status).
It seems to me that information that is deceiving is not worth showing (battery, tamper so on), so I made my own `fz.byun_smoke_on`
Also to process the smoke sensor restore message I made `fz.byun_smoke_off`

Reply:
Thanks for looking into this.